commit 0fb88a38a8f51d5735aba269b5f7ce92a61aebed (refs/changes/44/1144/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2016-12-19T11:24:48-08:00 (2 years, 10 months ago)
    
    DOCKER-987 Update docker build to allow container package selection from a label

diff --git a/bin/docker-build.js b/bin/docker-build.js
index 8071b6d..4f4fbdf 100644
--- a/bin/docker-build.js
+++ b/bin/docker-build.js
@@ -203,6 +203,14 @@ function sendMessage(message, opts, callback) {
 }
 
 
+/**
+ * Remove joyent specific labels that we don't want inside the built image.
+ */
+function sanatizeLabels(labels) {
+    delete labels['com.joyent.package'];
+}
+
+
 function downloadContext(opts) {
     var log = opts.log;
     var socket = opts.contextSocket;
@@ -364,13 +372,18 @@ function buildFromContext(opts, callback) {
         }
     });
 
+    // Sanitize labels.
+    var labels = JSON.parse(opts.payload.labels || '{}');
+    sanatizeLabels(labels);
+    labels = JSON.stringify(labels);
+
     var buildOpts = {
         buildargs: opts.payload.buildargs,
         commandType: 'build',
         containerRootDir: path.join('/zones', opts.uuid, 'root'),
         contextFilepath: opts.contextFilepath,
         dockerfile: opts.payload.dockerfile,
-        labels: opts.payload.labels,
+        labels: labels,
         suppressSuccessMsg: true,  // Stop the 'Build successful: ' message.
         existingImages: opts.payload.allDockerImages,
         log: opts.log,
@@ -655,6 +668,13 @@ function commitImage(opts, callback) {
                 if (opts.payload.comment) {
                     img.comment = opts.payload.comment;
                 }
+                // Sanatize labels.
+                if (img.config && img.config.Labels) {
+                    sanatizeLabels(img.config.Labels);
+                }
+                if (img.container_config && img.container_config.Labels) {
+                    sanatizeLabels(img.container_config.Labels);
+                }
                 cb();
             },
             function importImages(cb) {
